ssl/ssl_lib.c

```diff
@@ -972,6 +972,9 @@ SSL *ossl_ssl_connection_new_int(SSL_CTX *ctx, const SSL_METHOD *method)
     s->ext.ech.innerch1 = NULL;
     s->ext.ech.encoded_innerch = NULL;
     s->ext.ech.kepthrr = NULL;
+#endif
+#ifndef OPENSSL_NO_SECH
+    s->sech.symmetric_key = ctx->sech.symmetric_key;
```

Comment:
Needs to be a deep copy:
1. allocate memory for `s->sech.symmetric_key`
2. deep copy `ctx->sech.symmetric_key` to `s->sech.symmetric_key`
3. find the appropriate place to `OPENSSL_free` the connection's copy of the `sech.symmetric_key`